### ISSUE TITLE
Flag to remove css files when generating component!

### ISSUE THREAD

**Issue by IamManchanda** [angular/angular-cli#5890]
Sorry for [Duplicating](https://github.com/angular/angular/issues/15844) but had no choice 

---------------------------------------------------------------------------------------------------------------------

**I'm submitting an Advancement** 

Like many others i am a fan of global styles (ofcourse, if used correctly like [BEMIT](https://csswizardry.com/2015/08/bemit-taking-the-bem-naming-convention-a-step-further/) and i basically (quite surely like many others) don't like this below thing for my css unless needed

```scss
h3[_ngcontent-c0] {
  margin-top: 1rem;
  color: #0277bd;
}
```
I will instead use global styles by either using `styles.css` global file or using sass tools for my css so i basically delete my `foo.component.css` file and its declaration from a **foo** component and do it like this 

```scss
.c-foo--title {
  margin-top: 1rem;
  color: #0277bd;
}
```
In short, just would like to separate my styling with **Javascript and the HTML DOM**!

So just want to know if is there any way (and if not can it be provided) to have a flag on not adding separate component styling css file when we generate a component from CLI!

Something like 

```bash
$ ng g c --nocss foo
```

-------------------------------------------------------------------------------------------------------------------------------

**What is the motivation / use case for changing the behavior?**

- [BEMIT](https://csswizardry.com/2015/08/bemit-taking-the-bem-naming-convention-a-step-further/) 

**Please tell us about your environment:**

* **Angular version:** 4.0.X

 
* **Language:** TypeScript X.X 

* **Tool:** CLI

**Comment by grizzm0** [comment]
I'm not sure if there's a way to completly skip style generation, but you can pass `--inline-style` or `--is` to prevent the file from being generated. This will instead create a style key in the component annotation.

You can make this the default behaviour by changing angular-cli.json. 
See: https://github.com/angular/angular-cli/blob/master/packages/%40angular/cli/lib/config/schema.json

**Comment by sumitarora** [comment]
Closing as above.

**Comment by IamGiel** [comment]
i know this is closed.  Please advise though as to how I would do the opposite.  My current tech stack, does not allow for css to be created when using `ng g c <mycomponent>` it generates html, ts, and spec (which I omit using --spec=false).  I would like it to have mycomponent.route.ts and mycomponent.css.  Please advise - thanks

**Comment by StefanKern** [comment]
@grizzm0 it is `-is` ;)

**Comment by grizzm0** [comment]
@StefanKern Probably was `--is` two years ago.. :p
